.github/workflows/on-merge.yml

```diff
@@ -1,10 +1,11 @@
 name: Self-hosted runner (push to "main")
 
 on:
+  workflow_run:
+    workflows: ["Check for dependency modification"]
+    branches: ["main"]
+    types: [completed]
```

Comment:
I was thinking that `completed` means the CI will be run only if the workflow `Check for dependency modification` is successful.

After reading this [doc](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow), it is not the case. And this is good, as we want to run the push CI anyway (even if something wrong in the setup check).

Reply:
Yup exactly